### PR TITLE
feat: config scheme for ios and android

### PR DESCRIPTION
**Pull request by loido**

### CODE REVIEW

**Review by vunguyendhSTS — APPROVED** [review]

**Comment by baonguyenhsts** [comment]
Did you try to install the app on your simulator/device and check there two apps on your device screen?

**Comment by loido** [comment]
> Did you try to install the app on your simulator/device and check there two apps on your device screen?

yes, i did
image

**Review by baonguyenhsts — APPROVED** [review]

**Comment by huydosgtech** [comment]
wrong typo scheme on commit message

File: template/ios/Podfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ require_relative '../node_modules/@react-native-community/cli-platform-ios/nativ
 platform :ios, '12.4'
 install! 'cocoapods', :deterministic_uuids => false
 
-target 'RNBaseProject' do
+abstract_target 'RNBaseProjectCommonPods' do
```
Comment (ThinhKimVo) [inline template/ios/Podfile:7]: don't need, because app can config `pod install` for all env

File: template/ios/RNBaseProject dev-Info.plist [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ThinhKimVo) [inline template/ios/RNBaseProject dev-Info.plist:1]: only one .plist file. Can update config on env

File: template/ios/RNBaseProject.xcodeproj/xcshareddata/xcschemes/RNBaseProject.xcscheme [hunk 3]
```diff
@@ -1,10 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
    LastUpgradeVersion = "1210"
-   version = "1.3">
+   version = "1.7">
```
Comment (ThinhKimVo) [inline template/ios/RNBaseProject.xcodeproj/xcshareddata/xcschemes/RNBaseProject.xcscheme:4]: Purpose of upgrade to v 1.7?

File: template/package.json [hunk 4]
```diff
@@ -12,7 +12,13 @@
...
     "prepare": "husky install",
     "postinstall": "bash ./scripts/post-install.sh",
-    "reverse": "adb reverse tcp:8081 tcp:8081"
+    "reverse": "adb reverse tcp:8081 tcp:8081",
+    "android:staging": "react-native run-android --variant=stagingDebug",
+    "android:staging-release": "react-native run-android --variant=stagingRelease",
+    "android:dev": "react-native run-android --variant=developmentDebug",
+    "android:dev-release": "react-native run-android --variant=developmentRelease",
+    "android:prod": "react-native run-android --variant=productionDebug",
+    "android:prod-release": "react-native run-android --variant=productionRelease"
```
Comment (tuledu) [inline template/package.json:21]: How about running ios by script? @loido

**Comment by tuledu** [comment]
I think you should create other schemes instead of creating targets on ios @loido

File: template/ios/RNBaseProject dev-Info.plist (hunk 2 above)
Comment (loido) [reply template/ios/RNBaseProject dev-Info.plist]: I've updated

File: template/package.json (hunk 4 above)
Comment (loido) [reply template/package.json]: I've updated

File: template/ios/RNBaseProject.xcodeproj/xcshareddata/xcschemes/RNBaseProject.xcscheme (hunk 3 above)
Comment (loido) [reply template/ios/RNBaseProject.xcodeproj/xcshareddata/xcschemes/RNBaseProject.xcscheme]: I've updated

**Comment by honghoangsts** [comment]
Deadline 17/12/2022

File: template/android/app/build.gradle [hunk 5]
```diff
@@ -141,27 +141,18 @@ def reactNativeArchitectures() {
...
     ndkVersion rootProject.ext.ndkVersion
     compileSdkVersion rootProject.ext.compileSdkVersion
-
+    
     // add this block
     flavorDimensions "default"
     productFlavors {
         development {
-            minSdkVersion rootProject.ext.minSdkVersion
             applicationId "com.saigontechnolgy.rnbaseproject.development"
```
Comment (baonguyenhsts) [inline template/android/app/build.gradle:149]: put applicationId to .env

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: template/android/app/build.gradle (hunk 5 above)
Comment (loido) [reply template/android/app/build.gradle]: I've updated

**Review by tuledu — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rn-base-project",
-  "version": "0.0.3",
+  "version": "0.0.5",
   "main": "index.js",
   "repository": "https://github.com/saigon-technology/rn-base-project.git",
   "author": "React Native Teeam <SaiGon Technology Solutions>",
```
Comment (honghoangsts) [inline package.json:6]: Why you update this file. please rebase from branch `develop` or find root cause for this changed

File: template/android/app/src/development/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<resources>
+    <string name="app_name">BaseDev</string>
+    <string moduleConfig="true" name="CodePushDeploymentKey">DeploymentKey</string>
```
Comment (honghoangsts) [inline template/android/app/src/development/res/values/strings.xml:3]: Could you try to config it using env file.

File: template/android/app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -5,14 +5,14 @@
 
     <application
       android:name=".MainApplication"
-      android:label="@string/app_name"
+      android:label="@string/APP_NAME"
       android:icon="@mipmap/ic_launcher"
-      android:roundIcon="@mipmap/ic_launcher_round"
```
Comment (honghoangsts) [inline template/android/app/src/main/AndroidManifest.xml:10]: Why you remove round icon?

File: template/ios/RNBaseProject/Info.plist [hunk 9]
```diff
@@ -22,6 +22,8 @@
 	<string>????</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
+	<key>CodePushDeploymentKey</key>
+	<string>CodePushDeploymentKey</string>
```
Comment (honghoangsts) [inline template/ios/RNBaseProject/Info.plist:26]: Could you config to read it from env file

File: template/android/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -1,3 +1,4 @@
 <resources>
-    <string name="app_name">RNBaseProject</string>
+    <string name="app_name">Base</string>
+    <string moduleConfig="true" name="CodePushDeploymentKey">DeploymentKey</string>
```
Comment (honghoangsts) [inline template/android/app/src/main/res/values/strings.xml:3]: same above

File: template/android/app/src/staging/res/values/strings.xml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<resources>
+    <string name="app_name">BaseStg</string>
+    <string moduleConfig="true" name="CodePushDeploymentKey">DeploymentKey</string>
```
Comment (honghoangsts) [inline template/android/app/src/staging/res/values/strings.xml:3]: same above

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (loido) [reply package.json:6]: I fixed it to the original version

File: template/android/app/src/development/res/values/strings.xml (hunk 7 above)
Comment (loido) [reply template/android/app/src/development/res/values/strings.xml]: i've updated

File: template/android/app/src/main/AndroidManifest.xml (hunk 8 above)
Comment (loido) [reply template/android/app/src/main/AndroidManifest.xml]: i've updated

File: template/android/app/src/main/res/values/strings.xml (hunk 10 above)
Comment (loido) [reply template/android/app/src/main/res/values/strings.xml]: i've updated

File: template/android/app/src/staging/res/values/strings.xml (hunk 11 above)
Comment (loido) [reply template/android/app/src/staging/res/values/strings.xml]: i've updated

File: template/ios/RNBaseProject/Info.plist (hunk 9 above)
Comment (loido) [reply template/ios/RNBaseProject/Info.plist]: i've updated

File: package.json (hunk 6 above)
Comment (honghoangsts) [reply package.json]: Could you make sure you get the latest code.

**Comment by honghoangsts** [comment]
Please resolve the conflict and make sure everything you did can init as a template.

**Comment by loido** [comment]
>

File: template/package.json [hunk 12]
```diff
@@ -16,6 +16,15 @@
...
     "reverse": "adb reverse tcp:8081 tcp:8081",
+    "android:stg": "react-native run-android --variant=stagingDebug",
+    "android:stg-release": "react-native run-android --variant=stagingRelease",
+    "android:dev": "react-native run-android --variant=developmentDebug",
+    "android:dev-release": "react-native run-android --variant=developmentRelease",
+    "android:prod": "react-native run-android --variant=productionDebug",
+    "android:prod-release": "react-native run-android --variant=productionRelease",
+    "ios:prod": "react-native run-ios --scheme 'RNBaseProject'",
+    "ios:dev": "react-native run-ios --scheme 'RNBaseProjectDev'",
+    "ios:staging": "react-native run-ios --scheme 'RNBaseProjectStg'",
```
Comment (baonguyenhsts) [inline template/package.json:27]: Please create two scripts and add a function to pass variant (android), scheme (ios)

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: template/package.json (hunk 12 above)
Comment (loido) [reply template/package.json]: I've updated it, please review it again

**Review by baonguyenhsts — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]

**Review by anhle10051996 — APPROVED** [review]

File: docs/config-scheme.md [hunk 13]
```diff
@@ -0,0 +1,377 @@
...
+- To change app name, open file and rename
+
+`android/app/src/main/AndroidManifest.xml`
+
+```bash
+    <application
+      android:name=".MainApplication"
+      android:label="@string/APP_NAME" // change this line
+      android:icon="@mipmap/ic_launcher"
+      android:roundIcon="@mipmap/ic_launcher"
```
Comment (honghoangsts) [inline docs/config-scheme.md:169]: Why the value for roundIcon is ic_launcher

**Review by honghoangsts — APPROVED** [review]

File: docs/config-scheme.md (hunk 13 above)
Comment (loido) [reply docs/config-scheme.md]: i've updated it

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**